homeassistant/components/wemo.py

```diff
@@ -62,7 +66,7 @@ def stop_wemo(event):
     def discovery_dispatch(service, discovery_info):
         """Dispatcher for WeMo discovery events."""
         # name, model, location, mac
-        model_name = discovery_info.get('model_name')
+        component = discovery_info.get('component')
```

Comment:
Wait, maybe we still need to look for model name when the device is found by network discovery. Sorry, I forgot about that. You could first check for component in the discovery info, if not found, check for model name, and if model name do lookup.

Reply:
Ok, let's revert the last commit, that I suggested. That will make discovery info consistent. We can afford two lookups.